src/reducers/characterReducer.js

```diff
@@ -182,7 +182,21 @@ function characterReducer(state = initialState, action) {
         characters: [...state.characters, action.newCharacter],
       }
     }
+    case characterActions.EDIT_CHARACTER: {
+      const characterIndex = state.characters.findIndex((character) => {
+        return character.id === action.updatedCharacter.id
+      })
 
+      if (characterIndex === -1) return
+
+      const newCharactersArray = state.characters
+      newCharactersArray[characterIndex] = action.updatedCharacter
+
+      return {
+        ...state,
+        characters: [...newCharactersArray],
+      }
```

Comment:
This isn't how you should be updating your array, please make a new array, then update the item at the specified index in that new array